scripts/cluster.js

```diff
@@ -79,17 +79,25 @@ Cluster.prototype._replicateRawDB = function (sourceDB, targetDB) {
   });
 };
 
+
 Cluster.prototype._replicateDB = function (sourceDB, targetDB) {
   var self = this;
-  self._log('beginning replication of ' + sourceDB + '...');
-  return self._createDBIfMissing(targetDB).then(function () {
-    // Replicate security first so that security is put in place before data is copied over
-    return self._replicateSecurity(sourceDB, targetDB);
-  }).then(function () {
-    return self._replicateRawDB(sourceDB, targetDB);
-  }).then(function () {
-    self._log('finished replicating ' + sourceDB);
-  });
+  return this._sourceSlouch.db.exists(sourceDB)
+    .then(function (value) {
+      if (value === true) {
+        self._log('beginning replication of ' + sourceDB + '...');
```

Comment:
To make this easier to test I'd recommend creating a new method `Cluster.prototype._createAndReplicateDB()` and move in lines 88-96. Then call `Cluster.prototype._createAndReplicateDB()` in `Cluster.prototype._replicateDB()`. I'll mention the details of the test below.